Paracha_Junaid/Assignments/Flask_SQL/login_Val/server.py

```diff
@@ -0,0 +1,152 @@
+                flash("Bad email/pass combination")
+                valid = False
+                return redirect('/')
+
+            if valid == True :           
+                return redirect('/logged_in')
+        else:  
+            flash("You are not a Registered user please register")
+            session['Login_class'] = 'hide'
+            session['Register_class'] = 'show'
+            valid = False
+            print 'point2'
+            return render_template('index.html')     
+@app.route('/logged_in')
+def logged_in():
+    if not 'user' in session:
+        flash('Please login')
+        redirect('/')
+
+    return render_template('user.html')
+
+
+@app.route('/logout')
+def logout():
+    
+    return redirect('/')
+
+@app.route('/register', methods = ['POST'])
+def register():
+    Reg_valid = True
+    if len(request.form['email']) < 1 or len(request.form['first_name']) < 1 or len(request.form['last_name']) < 1:
+        flash("Fields cannot be blank!")
+        Reg_valid = False
+        return redirect('/')
+    elif not EMAIL_REGEX.match(request.form['email']):
+        flash("Invalid Email Address!")
+        Reg_valid = False
+        return redirect('/')
+    elif not request.form['first_name'].isalpha():
+        flash("Invalid Name! First Name cannot contain and characters or numbers")
+        Reg_valid = False
+        return redirect('/')
+    elif not request.form['last_name'].isalpha():
+        flash("Invalid Name! Last Name cannot contain and characters or numbers")
+        Reg_valid = False
+        return redirect('/')
+    elif not PW_REGEX.match(request.form['password']):
+        print request.form['password']
+        print type(request.form['password'])
+        flash("Invalid password!password must have at least 1 uppercase letter and 1 numeric value")
+        Reg_valid = False
+        return redirect('/')
+    elif request.form['password'] != request.form['c_password']:
+        flash('Password does not match the confirmation')
+        Reg_valid = False
+        return redirect('/')
+    if Reg_valid == True:
+       
+        pwd = bcrypt.generate_password_hash(request.form['password'])
+        Str = "INSERT INTO user(first_name, last_name, password, email, created_at, updated_at) VALUE (:first_name, :last_name, :password, :user_email, NOW(), NOW())"
+        data = { 
+                'first_name': request.form['first_name'] ,
+                'last_name': request.form['last_name'] ,
+                'password': pwd ,
+                'user_email': request.form['email'] 
+                }
+        try:
+            print mysql.query_db(Str,data)
+        except Exception as e:
+            print e
+            flash("Hey, there was a problem creating that account")
+            return redirect('/')
+        
+        session['added_email'] = request.form['email']
+        return redirect('/')
+    return redirect('/')
```

Comment:
need to remove all of the return redirects inside of the validations.  As soon the first failed validation hits it'll return without checking any of the other validations